bdk_electrum_example/src/main.rs

```diff
@@ -0,0 +1,189 @@
+mod electrum;
+
+use std::{collections::BTreeMap, fmt::Debug};
+
+use bdk_core::{
+    bitcoin::Network,
+    sparse_chain::{ChainIndex, SparseChain},
+};
+use bdk_keychain::{KeychainChangeSet, KeychainTracker};
+use electrum::ElectrumClient;
+
+use bdk_cli::{
+    anyhow::{self, anyhow, Context, Result},
+    clap::{self, Subcommand},
+};
+use log::debug;
+
+use electrum_client::ElectrumApi;
+
+#[derive(Subcommand, Debug, Clone)]
+enum ElectrumCommands {
+    /// Scans the addresses in the wallet using esplora API.
+    Scan {
+        /// When a gap this large has been found for a keychain it will stop.
+        #[clap(long, default_value = "5")]
+        stop_gap: usize,
+    },
+    /// Scans particular addresses using esplora API
+    Sync {
+        /// Scan all the unused addresses
+        #[clap(long)]
+        unused: bool,
+        /// Scan the script addresses that have unspent outputs
+        #[clap(long)]
+        unspent: bool,
+        /// Scan every address that you have derived
+        #[clap(long)]
+        all: bool,
+    },
+}
+
+fn complete_update<K: Debug + Clone + Ord, I: ChainIndex>(
+    new_sparsechain: &SparseChain<I>,
+    tracker: &KeychainTracker<K, I>,
+    client: &ElectrumClient,
+) -> Result<KeychainChangeSet<K, I>> {
+    let sparsechain_changeset = match tracker.chain().determine_changeset(new_sparsechain) {
+        Ok(cs) => cs,
+        Err(update_failure) => {
+            return Err(anyhow!(
```

Comment:
just use `Anyhow::Context::context` to add these annotations rather than matching them.

Reply:
The major reason I couldn't apply `context` or just pass on the error is the error type returned here isn't `std::err:Error`. So compiler is complaining on trait bound.. That's the reason for such awkward error handling.. Same with other places where the return type isn't `std::Error`..  

Here's the full error
```
error[E0599]: the method `context` exists for enum `Result<(), (KeychainChangeSet<Keychain, _>, HashSet<bdk_core::bitcoin::Txid>)>`, but its trait bounds were not satisfied
   --> bdk_electrum_example/src/main.rs:116:57
    |
116 |             tracker.apply_changeset(keychain_changeset).context("Changeset Application to tracker failed")?;
    |                                                         ^^^^^^^ method cannot be called on `Result<(), (KeychainChangeSet<Keychain, _>, HashSet<bdk_core::bitcoin::Txid>)>` due to unsatisfied trait bounds
    |
   ::: /home/raj/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs:504:1
    |
504 | pub enum Result<T, E> {
    | --------------------- doesn't satisfy `_: bdk_cli::anyhow::Context<(), (KeychainChangeSet<Keychain, _>, HashSet<bdk_core::bitcoin::Txid>)>`
    |
    = note: the following trait bounds were not satisfied:
            `(KeychainChangeSet<Keychain, _>, HashSet<bdk_core::bitcoin::Txid>): anyhow::context::ext::StdError`
            which is required by `Result<(), (KeychainChangeSet<Keychain, _>, HashSet<bdk_core::bitcoin::Txid>)>: bdk_cli::anyhow::Context<(), (KeychainChangeSet<Keychain, _>, HashSet<bdk_core::bitcoin::Txid>)>`
```
The work around for that is either this awkward handling, or create dedicated error type to return those errors. I think we should do the later, but would prefer to do them in a separate PR..